Tests/SentryTests/Helper/SentrySerializationTests.swift

```diff
@@ -298,6 +298,20 @@ class SentrySerializationTests: XCTestCase {
         XCTAssertEqual(payloadAsString.data(using: .utf8), item.data)
     }
     
+    /// This is to prevent https://github.com/getsentry/sentry-cocoa/issues/4280
+    /// With 8.33.0, writing an envelope could fail in the middle of the process, which the envelope
+    /// payload below simulates. The JSON stems from writing an envelope to disk with vast data
+    /// that leads to an OOM termination on v 8.33.0.
+    /// Running this test on v 8.33.0 leads to a crash.
+    func testEnvelopeWithDataCorruptEnvelope() throws {
+        let itemData = """
+                       {"event_id":"1990b5bc31904b7395fd07feb72daf1c","sdk":{"name":"sentry.cocoa","version":"8.33.0"}}
+                       {"type":"test","length":50}
+                       """.data(using: .utf8)!
+        
+        XCTAssertNil(SentrySerialization.envelope(with: itemData))
+    }
```

Comment:
h: This is a duplicate of `testReturnNilForCorruptedEnvelope`. Does it make sense to have both?

Reply:
`testReturnNilForCorruptedEnvelope` is slightly different. The contents of the envelope contain only an event but it's truncated by 1 byte. This one here has no attachment although there should be one. I think it adds value.

Reply:
From a code flow perspective, it’s exactly the same. The code will attempt to read the next 50 bytes and will not succeed. However, having two tests doesn’t cause any harm other than extending CI time, so I’m fine with it.

Reply:
I moved it now to SentrySDKTests to start the SDK with a corrupted envelope. Thanks for pointing it out.